### PR TITLE
Test Cases for Authenticator

### PR DESCRIPTION
**Pull request by RenukaFernando**
### Proposed changes in this pull request

- Test cases for Utils, Service, Internals

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/test/java/org/wso2/carbon/apimgt/rest/api/authenticator/internal/ServiceReferenceHolderTestCase.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package org.wso2.carbon.apimgt.rest.api.authenticator.internal;
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/test/java/org/wso2/carbon/apimgt/rest/api/authenticator/internal/ServiceReferenceHolderTestCase.java]: License headers missing

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/test/java/org/wso2/carbon/apimgt/rest/api/authenticator/utils/AuthUtilTestCase.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
+package org.wso2.carbon.apimgt.rest.api.authenticator.utils;
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/test/java/org/wso2/carbon/apimgt/rest/api/authenticator/utils/AuthUtilTestCase.java]: License headers missing

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/utils/AuthUtil.java [hunk 3]
```diff
@@ -162,23 +155,19 @@ public static String extractTokenFromHeaders(Request request, String cookieHeade
      * @return Cookie object.
      */
     public static NewCookie cookieBuilder(String name, String value, String path, boolean isSecure,
-                                          boolean isHttpOnly, String expiresIn) {
+                                          boolean isHttpOnly, String expiresIn, String environmentName) {
         StringBuilder stringBuilder = new StringBuilder();
-        stringBuilder.append(value).append(COOKIE_PATH_SEPERATOR).append(path).append(COOKIE_VALUE_SEPERATOR);
+        stringBuilder.append(value).append(COOKIE_PATH_SEPARATOR).append(path).append(COOKIE_VALUE_SEPARATOR);
+        if (expiresIn != null && !expiresIn.isEmpty()) {
```
Comment (tharikaGitHub) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/utils/AuthUtil.java]: We can use  StringUtils.isEmpty(expiresIn) to check for both null and empty. You can replace the condition with ! StringUtils.isEmpty(expiresIn)

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/test/java/org/wso2/carbon/apimgt/rest/api/authenticator/AuthenticatorServiceTestCase.java [hunk 4]
```diff
@@ -184,36 +185,53 @@ public void testGetTokens() throws Exception {
...
         Mockito.when(keyManager.getNewAccessToken(Mockito.any())).thenReturn(tokenInfo);
-        AccessTokenInfo tokenInfoResponseForJWTGrant = authenticatorService.getTokens("store",
-                "urn:ietf:params:oauth:grant-type:jwt-bearer", null, null, null, 0, null, "xxx-assertion-xxx");
-        Assert.assertEquals(tokenInfoResponseForJWTGrant, tokenInfo);
+        try {
+            AccessTokenInfo tokenInfoResponseForInvalidJWTGrant = authenticatorService.getTokens("store",
+                    "urn:ietf:params:oauth:grant-type:jwt-bearer", null, null, null, 0, null, "xxx-assertion-xxx", identityProvider);
+            Assert.assertEquals(tokenInfoResponseForInvalidJWTGrant, tokenInfo);
+        } catch (APIManagementException e) {
+            Assert.assertEquals(e.getMessage(), "User John dose not exists in this environment.");
```
Comment (tharikaGitHub) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/test/java/org/wso2/carbon/apimgt/rest/api/authenticator/AuthenticatorServiceTestCase.java]: Small spelling mistake -- does

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/AuthenticatorAPI.java [hunk 5]
```diff
@@ -198,33 +207,42 @@ public Response logout(@Context Request request, @PathParam("appName") String ap
...
         String accessToken = AuthUtil
-                .extractTokenFromHeaders(request, AuthenticatorConstants.ACCESS_TOKEN_2);
+                .extractTokenFromHeaders(request, AuthenticatorConstants.ACCESS_TOKEN_2, environmentName);
+
         if (accessToken != null) {
             try {
                 KeyManager keyManager = APIManagerFactory.getInstance().getKeyManager();
                 SystemApplicationDao systemApplicationDao = DAOFactory.getSystemApplicationDao();
-                AuthenticatorService authenticatorService = new AuthenticatorService(keyManager, systemApplicationDao);
+
```
Comment (Shehanir) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/AuthenticatorAPI.java:228]: Do we need extra line here ?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/AuthenticatorAPI.java [hunk 6]
```diff
@@ -198,33 +207,42 @@ public Response logout(@Context Request request, @PathParam("appName") String ap
...
                 AuthenticatorConstants.LOGOUT_SERVICE_CONTEXT + AuthenticatorConstants.URL_PATH_SEPERATOR + appName;
         String restAPIContext;
+
         if (appContext.contains(AuthenticatorConstants.EDITOR_APPLICATION)) {
             restAPIContext = AuthenticatorConstants.REST_CONTEXT + AuthenticatorConstants.URL_PATH_SEPERATOR
                     + AuthenticatorConstants.PUBLISHER_APPLICATION;
         } else {
             restAPIContext = AuthenticatorConstants.REST_CONTEXT + appContext;
         }
+
```
Comment (Shehanir) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/AuthenticatorAPI.java:217]: Extra line

**Pull request merged**